source/DSCResources/Resources/Vsphere.VmHostAdvancedSettings.ps1

```diff
@@ -0,0 +1,24 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$rules = $stig.RuleList | Select-Rule -Type 'VsphereAdvancedSettingsRule'
+
+$advancedSettings = @{}
+foreach ($rule in $rules)
+{
+    $key, $value = $rule.AdvancedSettings -split ' = '
```

Comment:
What is going on here?  It looks like we split on ' = ' and then we test if it's IsNullOrEmpty, which will return false (bool) if it's not and then we compare it to the value we just split?  And that value ($rule.AdvancedSettings) isn't a bool so how does the comparison work?  What am I missing?

Reply:
I helped Eric with this one, I think it should be:

``` PowerShell
if ([string]::IsNullOrEmpty($key) -eq $false)
```
The $rule.AdvancedSettings seems to be a typo that just happens to work.  However, I've corrected it.